diffkemp/semdiff/function_diff.py

```diff
@@ -282,8 +284,12 @@ def functions_diff(mod_first, mod_second,
                     fun_result = Result(fun_pair[2], fun_first, fun_second)
                 fun_result.first = fun_pair[0]
                 fun_result.second = fun_pair[1]
-                if fun_result.kind == Result.Kind.NOT_EQUAL:
-                    if fun_result.first.diff_kind in ["function", "type"]:
+                if fun_result.kind == Result.Kind.NOT_EQUAL or \
+                   config.full_diff:
+                    if fun_result.first.diff_kind in ["function", "type"] or (
+                            config.full_diff and
+                            fun_result.first.filename is not None and
+                            fun_result.second.filename is not None):
```

Comment:
IIUC, this part of the condition adds some syntactic differences (i.e. macros diffs) which were not previously shown. Does that make sense here? Shouldn't we just keep the condition as it was since `config.full_diff` is already checked above?

Reply:
You are right. For some reason, I thought that `diff_kind` is `None` when the compared functions are semantically equal, but that is not true.